Error while checking build/O2/fullCI for 6b75f0f5b21eb22a35ad90bba16e640a839d150b at 2024-12-10 02:01:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/SOURCES/O2/13787-slc8_x86-64/0/Detectors/AOD/src/AODProducerWorkflowSpec.cxx:385:23: error: 'struct o2::aodproducer::AODProducerWorkflowDPL::TrackQA' has no member named 'dTofdY'; did you mean 'dTofdX'?
/sw/SOURCES/O2/13787-slc8_x86-64/0/Detectors/AOD/src/AODProducerWorkflowSpec.cxx:2605:24: error: no match for 'operator[]' (operand types are 'o2::dataformats::GlobalTrackID' and 'o2::dataformats::GlobalTrackID::Source')
/sw/SOURCES/O2/13787-slc8_x86-64/0/Detectors/AOD/src/AODProducerWorkflowSpec.cxx:2607:25: error: 'gloCopy' was not declared in this scope
/sw/SOURCES/O2/13787-slc8_x86-64/0/Detectors/AOD/src/AODProducerWorkflowSpec.cxx:2609:30: error: 'safeInt8Clamp' was not declared in this scope
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13787/6b75f0f5b21eb22a35ad90bba16e640a839d150b/build_O2_fullCI/pretty.html).

--- a/Detectors/AOD/include/AODProducerWorkflow/AODProducerWorkflowSpec.h
+++ b/Detectors/AOD/include/AODProducerWorkflow/AODProducerWorkflowSpec.h
@@ -433,6 +433,8 @@ class AODProducerWorkflowDPL : public Task
     int8_t dRefGloSnp{std::numeric_limits<int8_t>::min()};
     int8_t dRefGloTgl{std::numeric_limits<int8_t>::min()};
     int8_t dRefGloQ2Pt{std::numeric_limits<int8_t>::min()};
+    int8_t dTofdX{std::numeric_limits<int8_t>::min()};
+    int8_t dTofdZ{std::numeric_limits<int8_t>::min()};
   };
 
   // helper struct for addToFwdTracksTable()

--- a/Detectors/AOD/src/AODProducerWorkflowSpec.cxx
+++ b/Detectors/AOD/src/AODProducerWorkflowSpec.cxx
@@ -381,7 +381,10 @@ void AODProducerWorkflowDPL::addToTracksQATable(TracksQACursorType& tracksQACurs
     trackQAInfoHolder.dRefGloZ,
     trackQAInfoHolder.dRefGloSnp,
     trackQAInfoHolder.dRefGloTgl,
-    trackQAInfoHolder.dRefGloQ2Pt);
+    trackQAInfoHolder.dRefGloQ2Pt,
+    trackQAInfoHolder.dTofdY,
+    trackQAInfoHolder.dTofdZ
+    );
 }
 
 template <typename mftTracksCursorType, typename AmbigMFTTracksCursorType>
@@ -2597,6 +2600,15 @@ AODProducerWorkflowDPL::TrackQA AODProducerWorkflowDPL::processBarrelTrackQA(int
     trackQAHolder.tpcdEdxTot1R = uint8_t(tpcOrig.getdEdx().dEdxTotOROC1 * dEdxNorm);
     trackQAHolder.tpcdEdxTot2R = uint8_t(tpcOrig.getdEdx().dEdxTotOROC2 * dEdxNorm);
     trackQAHolder.tpcdEdxTot3R = uint8_t(tpcOrig.getdEdx().dEdxTotOROC3 * dEdxNorm);
+    ///
+    float_t scaleTOF=0;
+    if (contributorsGID[GIndex::Source::TOF].isIndexSet()) { // ITS-TPC-TRD-TOF, ITS-TPC-TOF, TPC-TRD-TOF, TPC-TOF
+      const auto& tofMatch = data.getTOFMatch(trackIndex);
+      const float qpt = gloCopy.getQ2Pt();
+      scaleTOF=std::sqrt(o2::aod::track::trackQAScaledTOF[0]*o2::aod::track::trackQAScaledTOF[0]+qpt*qpt*o2::aod::track::trackQAScaledTOF[1]*o2::aod::track::trackQAScaledTOF[1])/(2.*o2::aod::track::trackQAScaleBins);
+      trackQAHolder.dTofdX = safeInt8Clamp(tofMatch.getDXatTOF()/scaleTOF);
+      trackQAHolder.dTofdZ = safeInt8Clamp(tofMatch.getDZatTOF()/scaleTOF);
+    }
 
     // Add matching information at a reference point (defined by
     // o2::aod::track::trackQARefRadius) in the same frame as the global track
@@ -2641,6 +2653,7 @@ AODProducerWorkflowDPL::TrackQA AODProducerWorkflowDPL::processBarrelTrackQA(int
         trackQAHolder.dRefGloSnp = safeInt8Clamp(((itsCopy.getSnp() + tpcCopy.getSnp()) * 0.5f - gloCopy.getSnp()) * scaleGlo(2));
         trackQAHolder.dRefGloTgl = safeInt8Clamp(((itsCopy.getTgl() + tpcCopy.getTgl()) * 0.5f - gloCopy.getTgl()) * scaleGlo(3));
         trackQAHolder.dRefGloQ2Pt = safeInt8Clamp(((itsCopy.getQ2Pt() + tpcCopy.getQ2Pt()) * 0.5f - gloCopy.getQ2Pt()) * scaleGlo(4));
+        //
 
         if (O2_ENUM_TEST_BIT(mStreamerMask, AODProducerStreamerMask::TrackQA)) {
           (*mStreamer) << "trackQA"
@@ -2684,6 +2697,9 @@ AODProducerWorkflowDPL::TrackQA AODProducerWorkflowDPL::processBarrelTrackQA(int
                        << "trackQAHolder.dRefGloSnp=" << trackQAHolder.dRefGloSnp
                        << "trackQAHolder.dRefGloTgl=" << trackQAHolder.dRefGloTgl
                        << "trackQAHolder.dRefGloQ2Pt=" << trackQAHolder.dRefGloQ2Pt
+                       << "trackQAHolder.dTofdY=" << trackQAHolder.dTofdY
+                       << "trackQAHolder.dTofdZ=" << trackQAHolder.dTofdZ
+                       << "scaleTOF=" << scaleTOF
                        << "\n";
         }
       }

--- a/Detectors/AOD/src/AODProducerWorkflowSpec.cxx
+++ b/Detectors/AOD/src/AODProducerWorkflowSpec.cxx
@@ -2697,7 +2697,7 @@ AODProducerWorkflowDPL::TrackQA AODProducerWorkflowDPL::processBarrelTrackQA(int
                        << "trackQAHolder.dRefGloSnp=" << trackQAHolder.dRefGloSnp
                        << "trackQAHolder.dRefGloTgl=" << trackQAHolder.dRefGloTgl
                        << "trackQAHolder.dRefGloQ2Pt=" << trackQAHolder.dRefGloQ2Pt
-                       << "trackQAHolder.dTofdY=" << trackQAHolder.dTofdY
+                       << "trackQAHolder.dTofdX=" << trackQAHolder.dTofdX
                        << "trackQAHolder.dTofdZ=" << trackQAHolder.dTofdZ
                        << "scaleTOF=" << scaleTOF
                        << "\n";

--- a/Detectors/AOD/src/AODProducerWorkflowSpec.cxx
+++ b/Detectors/AOD/src/AODProducerWorkflowSpec.cxx
@@ -383,8 +383,7 @@ void AODProducerWorkflowDPL::addToTracksQATable(TracksQACursorType& tracksQACurs
     trackQAInfoHolder.dRefGloTgl,
     trackQAInfoHolder.dRefGloQ2Pt,
     trackQAInfoHolder.dTofdY,
-    trackQAInfoHolder.dTofdZ
-    );
+    trackQAInfoHolder.dTofdZ);
 }
 
 template <typename mftTracksCursorType, typename AmbigMFTTracksCursorType>
@@ -2601,13 +2600,13 @@ AODProducerWorkflowDPL::TrackQA AODProducerWorkflowDPL::processBarrelTrackQA(int
     trackQAHolder.tpcdEdxTot2R = uint8_t(tpcOrig.getdEdx().dEdxTotOROC2 * dEdxNorm);
     trackQAHolder.tpcdEdxTot3R = uint8_t(tpcOrig.getdEdx().dEdxTotOROC3 * dEdxNorm);
     ///
-    float_t scaleTOF=0;
+    float_t scaleTOF = 0;
     if (contributorsGID[GIndex::Source::TOF].isIndexSet()) { // ITS-TPC-TRD-TOF, ITS-TPC-TOF, TPC-TRD-TOF, TPC-TOF
       const auto& tofMatch = data.getTOFMatch(trackIndex);
       const float qpt = gloCopy.getQ2Pt();
-      scaleTOF=std::sqrt(o2::aod::track::trackQAScaledTOF[0]*o2::aod::track::trackQAScaledTOF[0]+qpt*qpt*o2::aod::track::trackQAScaledTOF[1]*o2::aod::track::trackQAScaledTOF[1])/(2.*o2::aod::track::trackQAScaleBins);
-      trackQAHolder.dTofdX = safeInt8Clamp(tofMatch.getDXatTOF()/scaleTOF);
-      trackQAHolder.dTofdZ = safeInt8Clamp(tofMatch.getDZatTOF()/scaleTOF);
+      scaleTOF = std::sqrt(o2::aod::track::trackQAScaledTOF[0] * o2::aod::track::trackQAScaledTOF[0] + qpt * qpt * o2::aod::track::trackQAScaledTOF[1] * o2::aod::track::trackQAScaledTOF[1]) / (2. * o2::aod::track::trackQAScaleBins);
+      trackQAHolder.dTofdX = safeInt8Clamp(tofMatch.getDXatTOF() / scaleTOF);
+      trackQAHolder.dTofdZ = safeInt8Clamp(tofMatch.getDZatTOF() / scaleTOF);
     }
 
     // Add matching information at a reference point (defined by

--- a/Framework/Core/include/Framework/DataTypes.h
+++ b/Framework/Core/include/Framework/DataTypes.h
@@ -130,7 +130,7 @@ constexpr std::array<float, 5> trackQAScaleContP0{0.257192, 0.0775375, 0.0042428
 constexpr std::array<float, 5> trackQAScaleContP1{0.189371, 0.409071, 0.00694444, 0.00720038, 0.0806902};
 constexpr std::array<float, 5> trackQAScaleGloP0{0.130985, 0.0775375, 0.00194703, 0.000405458, 0.0160007};
 constexpr std::array<float, 5> trackQAScaleGloP1{0.183731, 0.409071, 0.00621802, 0.00624881, 0.0418957};
-constexpr std::array<float, 2> trackQAScaledTOF{1.1,0.33};
+constexpr std::array<float, 2> trackQAScaledTOF{1.1, 0.33};
 } // namespace o2::aod::track
 
 namespace o2::aod::fwdtrack

--- a/Framework/Core/include/Framework/DataTypes.h
+++ b/Framework/Core/include/Framework/DataTypes.h
@@ -130,6 +130,7 @@ constexpr std::array<float, 5> trackQAScaleContP0{0.257192, 0.0775375, 0.0042428
 constexpr std::array<float, 5> trackQAScaleContP1{0.189371, 0.409071, 0.00694444, 0.00720038, 0.0806902};
 constexpr std::array<float, 5> trackQAScaleGloP0{0.130985, 0.0775375, 0.00194703, 0.000405458, 0.0160007};
 constexpr std::array<float, 5> trackQAScaleGloP1{0.183731, 0.409071, 0.00621802, 0.00624881, 0.0418957};
+constexpr std::array<float, 2> trackQAScaledTOF{1.1,0.33};
 } // namespace o2::aod::track
 
 namespace o2::aod::fwdtrack

